Xamarin.Forms.Platform.Android/CollectionView/CarouselViewAdapter.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using Android.Content;
+#if __ANDROID_29__
+using AndroidX.RecyclerView.Widget;
+#else
+using Android.Support.V7.Widget;
+#endif
+
+namespace Xamarin.Forms.Platform.Android.CollectionView
+{
+	public class CarouselViewAdapter<TItemsView, TItemsViewSource> : ItemsViewAdapter<TItemsView, TItemsViewSource>
+		where TItemsView : ItemsView
+		where TItemsViewSource : IItemsViewSource
+	{
+		protected readonly CarouselView CarouselView;
+		internal CarouselViewAdapter(CarouselView itemsView, Func<View, Context, ItemContentView> createItemContentView = null) : base(itemsView as TItemsView, createItemContentView)
+		{
+			CarouselView = itemsView;
+		}
+
+		public override int ItemCount => CarouselView.Loop ? int.MaxValue : ItemsSource.Count;
+
+		public override void OnBindViewHolder(RecyclerView.ViewHolder holder, int position)
+		{
+			int positionInList = CarouselView.Loop ? position % ItemsSource.Count : position;
```

Comment:
We should null check CarouselView and ItemsSource

Reply:
fixed